app/containers/filterContainer/filterSaveBox/index.tsx

```diff
@@ -0,0 +1,167 @@
+import * as React from "react";
+import { connect, Dispatch } from "react-redux";
+import { withRouter, RouteComponentProps } from "react-router";
+import { isEqual, findIndex } from "lodash";
+import Popper from "@material-ui/core/Popper";
+import ClickAwayListener from "@material-ui/core/ClickAwayListener";
+import { withStyles } from "../../../helpers/withStylesHelper";
+import { ArticleSearchState } from "../../../components/articleSearch/records";
+import { Filter } from "../../../api/member";
+import { setSavedFilterSet, putMyFilters } from "../../../components/articleSearch/actions";
+import PapersQueryFormatter from "../../../helpers/papersQueryFormatter";
+import getQueryParamsObject from "../../../helpers/getQueryParamsObject";
+import SavedFilterItem from "../savedFilterItem";
+import newFilterSetTitleGenerator from "../../../helpers/newFilterSetTitleGenerator";
+import FilterTitleBox from "./titleBox";
+import Icon from "../../../icons";
+const styles = require("./filterSaveBox.scss");
+
+interface FilterSaveBoxProps {
+  articleSearchState: ArticleSearchState;
+  dispatch: Dispatch<any>;
+}
+
+const FilterSaveBox: React.FunctionComponent<FilterSaveBoxProps & RouteComponentProps<any>> = props => {
+  const { articleSearchState, dispatch, history, location } = props;
+  const { savedFilterSet, myFilters, searchInput, sort } = articleSearchState;
+
+  let popoverAnchorEl: HTMLDivElement | null;
+
+  const [isOpen, setIsOpen] = React.useState(false);
+  const [isChange, setIsChange] = React.useState(false);
+
+  const rawQueryParamsObj = getQueryParamsObject(location.search);
+
+  React.useEffect(
+    () => {
+      const currentFilter = PapersQueryFormatter.objectifyPapersFilter(rawQueryParamsObj.filter);
+      const savedFilter = !!savedFilterSet
+        ? PapersQueryFormatter.objectifyPapersFilter(savedFilterSet.filter)
+        : PapersQueryFormatter.objectifyPapersFilter();
+
+      if (isEqual(savedFilter, currentFilter)) {
+        setIsChange(false);
+      } else {
+        setIsChange(true);
+      }
+    },
+    [props.location, savedFilterSet]
+  );
+
+  function handleClickSaveButton(changedFilterReq: Filter | string) {
```

Comment:
I think this function is too verbose and ambiguous. 
Please separate this function into modifying and create(save?) function.